### PR TITLE
Update Firebase Cloud Functions to work in Production

### PR DESCRIPTION
**Pull request by Division7**
In this PR, I updated the URLs that the cloud functions create to request against the UCSB api. In production, the url is provided with a `/` in front of the request, which does not appear in simulator. To account for this, it now slices the `/` off where appropriate, or stays in the poll function.

Can be verified with https://us-central1-goldtracker-beb96.cloudfunctions.net/search, a valid firebase ID token, and the parameters described by https://developer.ucsb.edu/content/academic-curriculums#/Classes/Classes_GetClassesAsync

Polling a particular course can be verified with https://us-central1-goldtracker-beb96.cloudfunctions.net/poll, a valid firebase ID token, and the correct `/quarter/enrollcode/` as described here https://developer.ucsb.edu/content/academic-curriculums#/Classes/Classes_GetClassesAsync

### CODE REVIEW

**Review by xinyao-song — APPROVED** [review]
LGTM (UwU

**Review by zhenbi93 — APPROVED** [review]
LGTM, after merging please check ClassRegister.js file again.

**Pull request merged**
